### PR TITLE
📖 update links in release tasks docs

### PR DESCRIPTION
**Pull request by cahillsf**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Relates to https://github.com/kubernetes-sigs/cluster-api/issues/9800

follow up from last v1.6 release cut

### CODE REVIEW

**Comment by cahillsf** [comment]
👋 @furkatgofurov7 did i capture all the links we wanted to update?

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -77,10 +77,12 @@ This document details the responsibilities and tasks for each role in the releas
 
 #### Finalize release schedule and team
 
-1. Finalize release schedule and team in the [docs/release/releases](../../docs/release/releases), e.g. [release-1.4.md](../../docs/release/releases/release-1.4.md).
+1. Finalize release schedule and team in the [docs/release/releases](../../docs/release/releases), e.g. [release-1.4.md](../../docs/release/releases/release-1.6.md).
```
Comment (furkatgofurov7) [inline docs/release/release-tasks.md:80]:
```suggestion
1. Finalize release schedule and team in the [docs/release/releases](../../docs/release/releases), e.g. [release-1.6.md](../../docs/release/releases/release-1.6.md).
```

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -121,6 +123,7 @@ Prior art:
 
 * 1.4 - https://github.com/kubernetes-sigs/cluster-api/pull/7692/files
 * 1.5 - https://github.com/kubernetes-sigs/cluster-api/pull/8430/files
+* 1.6 - https://github.com/kubernetes-sigs/cluster-api/pull/9097
```
Comment (furkatgofurov7) [inline docs/release/release-tasks.md:126]: Probably we can drop 1.4 from this list

File: docs/release/release-tasks.md [hunk 3]
```diff
@@ -397,7 +400,7 @@ Someone with access to Netlify should:
 The goal of this task is to ensure the quickstart has links to the latest `clusterctl` binaries.
 
 Update clusterctl links in the quickstart (on main and cherry-pick onto release-1.4).
```
Comment (furkatgofurov7) [inline docs/release/release-tasks.md:402]:
```suggestion
Update clusterctl links in the quickstart (on main and cherry-pick onto release-1.6).
```

**Review by furkatgofurov7 — COMMENTED** [review]
@cahillsf I would update all references in the doc for "Prior art" mentions with newer PR references and adapt the doc to be based off of the latest 1.6 release, i.e replacing all mentions of `release-1.4` with `release-1.6` and changes needed accordingly

**Review by willie-yao — COMMENTED** [review]
Looks good pending one nit!

File: docs/release/release-tasks.md [hunk 4]
```diff
@@ -236,12 +237,12 @@ to a newer Go minor version according to our [backport policy](./../../CONTRIBUT
...
-      * For an example please see: [PR: clusterctl 1.1.5](https://github.com/Homebrew/homebrew-core/pull/105075/files).
+      * For an example please see: [PR: clusterctl 1.5.3](https://github.com/Homebrew/homebrew-core/pull/152279).
       * Homebrew has [conventions for commit messages](https://docs.brew.sh/Formula-Cookbook#commit) usually
-        the commit message for us should look like: `clusterctl 1.1.5`.
-6. **For minor releases** Set EOL date for previous release (prior art: https://github.com/kubernetes-sigs/cluster-api/issues/7146) and update Cluster API support and guarantees in CONTRIBUTING.md (prior art: https://github.com/kubernetes-sigs/cluster-api/pull/8308).
+        the commit message for us should look like: `clusterctl 1.5.3`.
+6. **For minor releases** Set EOL date for previous release and update Cluster API support and guarantees in CONTRIBUTING.md (prior art: https://github.com/kubernetes-sigs/cluster-api/pull/9119/files).
```
Comment (furkatgofurov7) [inline docs/release/release-tasks.md:245]: The prior art PR looks incorrect here, 9119 should be referenced as prior art for https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#update-supported-versions task at the beginning of the cycle (l.325 in this document). Here you should use: https://github.com/kubernetes-sigs/cluster-api/pull/9817/files instead

**Review by furkatgofurov7 — COMMENTED** [review]
Few tasks from comms team seem to be still needing an update (first 3) and some minor nits, overall looks good so far

File: docs/release/release-tasks.md [hunk 5]
```diff
@@ -542,4 +543,4 @@ The goal of bug triage is to triage incoming issues and if necessary flag them w
 and add them to the milestone of the current release.
 
 We probably have to figure out some details about the overlap between the bug triage task here, release leads
-and Cluster API maintainers.
\ No newline at end of file
+and Cluster API maintainers.
```
[the remark this answers is not in the document]
Comment (cahillsf) [reply docs/release/release-tasks.md:546]: hmm, yeah that is strange... i think there was a new line added at some point that is being reflected down here?  any ideas for how to fix?

**Comment by cahillsf** [comment]
> Few tasks from comms team seem to be still needing an update (first 3) and some minor nits, overall looks good so far

thanks @furkatgofurov7 , i pushed an update to the first two. [this one](https://github.com/kubernetes-sigs/cluster-api/blob/9334e1813ad050ec7cd763c54563c9f669bc1b5b/docs/release/release-tasks.md#ensure-the-book-for-the-new-release-is-available)is a bit interesting because it doesn't look like they've been done for the v1.6 cycle.

 it's strange though because this link for 1.5 works (https://release-1-5.cluster-api.sigs.k8s.io/) even though there does not appear to be a cname entry for it in the file referenced for prior art: https://github.com/ykakarap/k8s.io/blob/5a3c7bfb63ca564a0912ee346a1585b02530b0d0/dns/zone-configs/k8s.io._0_base.yaml#L398-L400

cc @willie-yao 

any ideas as to why 1.5 link is served without a CNAME entry?  how can we move forward with having 1.6 link serve correctly?

File: docs/release/release-tasks.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (cahillsf) [reply docs/release/release-tasks.md]: ah, it was a newline at the end of the doc smh.. should be good now 😄

**Comment by willie-yao** [comment]
> any ideas as to why 1.5 link is served without a CNAME entry? how can we move forward with having 1.6 link serve correctly?

Hmm that is quite weird. I didn't work on this task for the v1.6 cycle but I'll take a look and work on a fix!

**Review by willie-yao — APPROVED** [review]
/lgtm

**Comment by furkatgofurov7** [comment]
> [this one](https://github.com/kubernetes-sigs/cluster-api/blob/9334e1813ad050ec7cd763c54563c9f669bc1b5b/docs/release/release-tasks.md#ensure-the-book-for-the-new-release-is-available) is a bit interesting because it doesn't look like they've been done for the v1.6 cycle.

why, we did it in: https://github.com/kubernetes-sigs/cluster-api/pull/9767/files & https://github.com/kubernetes/k8s.io/pull/6114. Generally, you can also take a look at https://github.com/kubernetes-sigs/cluster-api/issues/9094 for all the references (PRs) of the tasks

**Comment by cahillsf** [comment]
ah thanks @furkatgofurov7, smh ended up looking at a fork:
> https://github.com/ykakarap/k8s.io/blob/5a3c7bfb63ca564a0912ee346a1585b02530b0d0/dns/zone-configs/k8s.io._0_base.yaml#L398-L400

just to confirm:
- it is expected that https://release-1-6.cluster-api.sigs.k8s.io redirects to https://cluster-api.sigs.k8s.io/?
- for `release-1.6` it is expected to add the link for the *previous* minor release in the book intro? https://github.com/kubernetes-sigs/cluster-api/pull/9767/files?

**Comment by cahillsf** [comment]
cool thanks for all the input here -- @willie-yao @furkatgofurov7 mind giving another lgtm when you have a chance?  i think all the feedback has been addressed

**Review by furkatgofurov7 — COMMENTED** [review]
thanks @cahillsf, lgtm pending squash from my side

**Pull request merged**